src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -261,11 +261,12 @@ bool tryBindAsBuilderType([NotNullWhen(true)] out BoundInterpolatedString? resul
                     new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(baseStringLength), intType) { WasCompilerGenerated = true },
                     new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(numFormatHoles), intType) { WasCompilerGenerated = true });
 
-                BoundCall? createExpression = BindCreateCall(unconvertedInterpolatedString.Syntax, interpolatedStringBuilderType, arguments, diagnostics);
+                // PROTOTYPE(interp-string): Support optional out param for whether the builder was created successfully and passing in other required args
+                BoundExpression? createExpression = MakeClassCreationExpression(interpolatedStringBuilderType, arguments, unconvertedInterpolatedString.Syntax, diagnostics);
```

Comment:
>MakeClassCreationExpression [](http://example.com/codeflow?start=52&length=27)

BindCreateCall used to report an error unless it was producing a BoundCall node. Should we be doing something similar here? For BoundDynamicObjectCreationExpression, for example?

Reply:
I have a prototype comment in LocalLowering to handle dynamic at a later point. I'll add assert for now that it either HasErrors or is a constructor.